amr-wind/wind_energy/ABLBoundaryPlane.cpp

```diff
@@ -543,8 +556,9 @@ void ABLBoundaryPlane::populate_data(
 
     for (amrex::OrientationIter oit; oit; ++oit) {
         auto ori = oit();
-        if ((not m_in_data.is_populated(ori)) or
-            (fld.bc_type()[ori] != BC::mass_inflow))
+        if ((!m_in_data.is_populated(ori)) ||
+            (fld.bc_type()[ori] != BC::mass_inflow) ||
+            (lev >= m_in_data.nlevels(ori)))
```

Comment:
@marchdf If I understand the logic correctly, the code will silently do nothing if there is a refined level that touches the inlet boundary that was not part of the original _precursor simlation_. This would mean that the inflow boundary has wrong values for the finer levels. I don't think a subsequent `fillpatch` will fix this issue either as for inflow it expects the user code to set the boundary values. It might make sense to check for intersection of the finer level box with inflow and throw here?

Reply:
You understood correctly. And yes it probably makes more sense to do check like I do for outflow where I check the intersection of the fine level with the boundary and then throw.

Reply:
Ok I tried to address... but it's tricky. Lmk what you think.